### PR TITLE
Item Stacking Implemented. New PR cause last one, I messed up.

### PR DESCRIPTION
**Pull request by bwu113**

### CODE REVIEW

File: MapleServer2/Types/SkillTab.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
+using System.Diagnostics;
```
Comment (AlanMorel) [inline MapleServer2/Types/SkillTab.cs:3]: This looks like it can be removed?

File: MapleServer2/PacketHandlers/Game/FieldEnterHandler.cs [hunk 2]
```diff
@@ -45,70 +45,87 @@ public override void Handle(GameSession session, PacketReader packet)
...
+            };
+            InventoryController.Add(session, item);
+            InventoryController.Add(session, item2);
+            InventoryController.Add(session, item3);
+
+
+
+
+
+            // Add mail for testing
```
Comment (AlanMorel) [inline MapleServer2/PacketHandlers/Game/FieldEnterHandler.cs:74]: If we're going to comment this much code, can you use `/* */` instead? Also we don't need so many blank lines between the inventory code and the commented out mail code

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/SkillTab.cs (hunk 1 above)
Comment (bwu113) [reply MapleServer2/Types/SkillTab.cs:3]: This file is unrelated. don't worry about it. thats from me testing skill tree issue.

File: MapleServer2/Tools/XmlParser.cs [hunk 3]
```diff
@@ -30,14 +30,25 @@ public static SkillTab ParseSkills(int job, string name = "test")
...
-                skillTab.AddOrUpdate(new Skill(id, 1, 0, feature, sub));
+                if (DefaultSkill > 0)
+                {
+                    skillTab.AddOrUpdate(new Skill(id, 1, 1, feature, sub));
+                }
+                else
+                {
+                    skillTab.AddOrUpdate(new Skill(id, 1, 0, feature, sub));
+                }
+
```
Comment (rhydev) [inline MapleServer2/Tools/XmlParser.cs:51]: This shouldn't be part of this PR, or if it is part of it, it should be feature complete.

File: MapleServer2/Types/Player.cs [hunk 4]
```diff
@@ -84,7 +84,7 @@ public class Player
 
         public static Player Default(long accountId, long characterId, string name = "SparkmodF")
         {
-            int job = 10;
+            int job = 100;
```
Comment (rhydev) [inline MapleServer2/Types/Player.cs:87]: What's the purpose of this change?

File: MapleServer2/Types/SkillTab.cs (hunk 1 above)
Comment (rhydev) [inline MapleServer2/Types/SkillTab.cs:3]: Still shouldn't be here

File: MapleServer2/Types/SkillTab.cs [hunk 5]
```diff
@@ -33,7 +34,6 @@ public void AddOrUpdate(Skill skill)
             {
                 Skills.Add(skill.Id, skill);
             }
-
```
Comment (rhydev) [inline MapleServer2/Types/SkillTab.cs:36]: Why is this new line removed?

File: MapleServer2/PacketHandlers/Game/FieldEnterHandler.cs [hunk 6]
```diff
@@ -45,29 +45,43 @@ public override void Handle(GameSession session, PacketReader packet)
...
+
             int[] catalysts = { 40100001, 40100002, 40100003, 40100021, 40100023, 40100024, 40100026 };
-            foreach (int catalyst in catalysts) {
-                var item = new Item(catalyst) { Amount = 99999, Uid = catalyst };
-                session.Inventory.Add(item);
-                session.Send(ItemInventoryPacket.Add(item));
-            }
-            */
 
-            // Add mail for testing
```
Comment (rhydev) [inline MapleServer2/PacketHandlers/Game/FieldEnterHandler.cs:57]: Please uncomment mail, it shouldn't interfere at all with inventory.

File: MapleServer2/Tools/InventoryController.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+                    i.Amount = i.Amount + item.Amount;
+                    session.Send(ItemInventoryPacket.Update(i.Uid, i.Amount));
+                    return;
+                }
+            }
+        }
+        session.Inventory.Add(item); // adds item into internal database
+        session.Send(ItemInventoryPacket.Add(item)); // sends packet to add item clientside.
+    }
+}
```
Comment (rhydev) [inline MapleServer2/Tools/InventoryController.cs:40]: There are many functions which remain unimplemented, update and remove for example. I believe these should be part of item stacking since it directly relates to the stacking of items.

**Review by rhydev — CHANGES_REQUESTED** [review]

File: MapleServer2/Types/Player.cs (hunk 4 above)
Comment (bwu113) [reply MapleServer2/Types/Player.cs:87]: These files aren't suppose to be in the pr.

File: MapleServer2/Tools/InventoryController.cs (hunk 7 above)
Comment (bwu113) [reply MapleServer2/Tools/InventoryController.cs:40]: Other than for splitting. I don't see any use in implementing those 2 methods until there is a natural way of obtaining items.

File: MapleServer2/Types/Player.cs (hunk 4 above)
Comment (bwu113) [reply MapleServer2/Types/Player.cs:87]: No purpose. ignore this file. Its part of the minor skill tree fix which will have its separate pr after this

**Review by rhydev — APPROVED** [review]
